packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx

```diff
@@ -143,6 +143,13 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
     const monthPickerOnly = !showMonthPickerAsOverlay && !isDayPickerVisible;
     const overlayedWithButton = showMonthPickerAsOverlay && showGoToToday;
 
+    let goTodayEnabled = showGoToToday;
+
+    if (goTodayEnabled && navigatedDayDate && selectedDate) {
+      goTodayEnabled =
+        navigatedDayDate.getFullYear() !== selectedDate.getFullYear() || navigatedDayDate.getMonth() !== selectedDate.getMonth();
```

Comment:
You probably meant to use "today" date rather than "selected" date.

Bug happens when you navigate to previous month, then select a day in the middle of the month, the go to today button gets disabled and it shouldn't

Reply:
related bug, you're using getMonth() but that doesn't cover what days are actually visible - if you select a visible day at the beginning or end of the range that's in the previous month, the button gets enabled again even if it can't do anything.